### PR TITLE
Test resolving a multisample renderbuffer keeps its contents.

### PR DESCRIPTION
**Pull request by greggman**
The Apple Metal backend had an issue that blitting a multisample
renderbuffer could lose its content.

### CODE REVIEW

File: sdk/tests/conformance2/renderbuffers/multisample-draws-between-blits.html [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+    };
+
+    debug('fill with red');
+    drawAndResolve([1, 0, 0, 1], [
+      2, 0, 0, 0,
+      0, 2, 0, 0,
+      0, 0, 1, 0,
+      -1, -1, 0, 1,
+    ]);
+    check(0, 0, w, h, [255, 0, 0, 255], 'whole thing');
```
Comment (kenrussell) [inline sdk/tests/conformance2/renderbuffers/multisample-draws-between-blits.html:141]: Could you define variables for the colors? e.g.

`const red = [255, 0, 0, 255];`

**Review by kenrussell — APPROVED** [review]
Great work on this test @greggman .

Are all of the draw phases strictly needed to reproduce the bug? If not, could you trim it down to the minimum needed?

Could you please edit the PR description to say:

1) What GPU(s) were affected by this bug
2) Please point to anglebug.com/ and the associated bug ID

Thanks. I'd like to take a quick look again before we merge this.

File: sdk/tests/conformance2/renderbuffers/multisample-draws-between-blits.html [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+
+    void main() {
+      gl_Position = mat * position;
+      texcoord = position.xy;
+    }
+    `;
+
+    const texFS = `#version 300 es
+    precision mediump float;
+  in vec2 texcoord;
```
[the remark this answers is not in the document]
Comment (greggman) [reply sdk/tests/conformance2/renderbuffers/multisample-draws-between-blits.html]: done

File: sdk/tests/conformance2/renderbuffers/multisample-draws-between-blits.html [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+<script src="../../js/js-test-pre.js"></script>
+<script src="../../js/webgl-test-utils.js"></script>
+</head>
+<body>
+<canvas id="canvas" width="128" height="64" style="width: 32px; height: 32px;"></canvas>
+<div id="description"></div>
+<div id="console"></div>
+<script>
+"use strict";
+var wtu = WebGLTestUtils;
```
[the remark this answers is not in the document]
Comment (greggman) [reply sdk/tests/conformance2/renderbuffers/multisample-draws-between-blits.html]: done

File: sdk/tests/conformance2/renderbuffers/multisample-draws-between-blits.html (hunk 1 above)
Comment (greggman) [reply sdk/tests/conformance2/renderbuffers/multisample-draws-between-blits.html]: done

**Comment by greggman** [comment]
shorted the number of draws.

**Comment by kenrussell** [comment]
Thanks. Which GPU(s) do you know reproduce the failures in this test? Which browsers? Also, which ANGLE bug is this associated with? Is it anglebug.com/7073 ?

**Comment by greggman** [comment]
That's in the PR description but copied and pasted here

The Apple Metal backend had an issue that blitting a multisample
renderbuffer could lose its content. Whether or not the contents
is lost is up to the driver. This repoed on M1 Macs.

See https://bugs.chromium.org/p/angleproject/issues/detail?id=7073

**Comment by kenrussell** [comment]
Thanks, I didn't realize the commit description was different from the first comment in this PR's thread.

**Pull request merged**